core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -87,29 +87,25 @@
  * <p>Instances of this class are not safe for use by multiple threads.</p>
  */
 public class Transaction extends ChildMessage {
+    private static final Comparator<Transaction> SORT_TX_BY_ID = Comparator.comparing(Transaction::getTxId);
     /**
      * A comparator that can be used to sort transactions by their updateTime field. The ordering goes from most recent
      * into the past.
      */
-    public static final Comparator<Transaction> SORT_TX_BY_UPDATE_TIME = (tx1, tx2) -> {
-        Instant t1 = tx1.getUpdateTimeInstant().orElse(Instant.EPOCH);
-        Instant t2 = tx2.getUpdateTimeInstant().orElse(Instant.EPOCH);
-        final int updateTimeComparison = -(t1.compareTo(t2));
-        //If time1==time2, compare by tx hash to make comparator consistent with equals
-        return updateTimeComparison != 0 ? updateTimeComparison : tx1.getTxId().compareTo(tx2.getTxId());
-    };
+    public static final Comparator<Transaction> SORT_TX_BY_UPDATE_TIME =
+            Comparator.<Transaction, Instant>comparing(
+                            tx -> tx.getUpdateTimeInstant().orElse(Instant.EPOCH),
+                            Comparator.reverseOrder())
+                    .thenComparing(SORT_TX_BY_ID);
```

Comment:
I would extract `tx -> tx.getUpdateTimeInstant().orElse(Instant.EPOCH)` to a private, static method. Maybe you can come up with a better name for the private method. I suppose the method could also be added as an instance method if that makes more sense.

```
    public static final Comparator<Transaction> SORT_TX_BY_UPDATE_TIME =
            Comparator.comparing(Transaction::txUpdateTimeOrElseEpoch, Comparator.reverseOrder())
                        .thenComparing(SORT_TX_BY_ID);
                        
    private static Instant txUpdateTimeOrElseEpoch(Transaction tx) {
        return tx.getUpdateTimeInstant().orElse(Instant.EPOCH);
    }
```

Reply:
Note that the extracted method eliminates the need for `<Transaction, Instant>`